docs/NRI.md

```diff
@@ -0,0 +1,93 @@
+extensions into OCI-compatible container runtimes. It provides basic
+mechanisms for plugins to track the state of containers and to make
+limited changes to their configuration.
+
+NRI itself is agnostic to the internal implementation details of any
+container runtime. It provides an adaptation library which runtimes
+use to integrate to and interact with NRI and plugins. In principle
+any NRI plugin should be able to work with NRI-enabled runtime.
+
+For a detailed description of NRI and its capabilities please take a
+look at the [NRI respository](https://github.com/containerd/nri).
+
+## Containerd NRI Integration
+
+NRI support in containerd is split into two parts both logically and
+physically. These parts are a common plugin (/nri/*) to integrate to
+NRI and CRI-specific bits (/pkg/cri/service/nri-api) of converting
+data between the runtime-agnostic NRI representation and the internal
+representation of the CRI plugin.
+
+### Containerd NRI Plugin
+
+The containerd common NRI plugin implements the core logic of integrating
+to and interacting with NRI. However, it does this without any knowledge
+about the internal representation of containers or pods within containerd.
+It defines an additional interface, Domain, which is used whenever the
+internal representation of a container or pod needs to be translated to
+the runtime agnostic NRI one, or when a configuration change requested by
+an external NRI plugin needs to be applied to a container within continerd.
+
+### NRI Support for CRI Containers
+
+The containerd CRI plugin registers itself as an above mentioned NRI
+Domain, to allow container configuration to be customized by external
+NRI plugins. Currently this Domain interface is only implemented for
+the original CRI `pkg/cri/server` implementation. Implementing it for
+the more recent experimental `pkg/cri/sbserver` implementation is on
+the TODO list.
+
+### NRI Support for Other Container 'Domains'
+
+The main reason for this split of functionality is that it should allow
+us to experiment with exposing other than just CRI containers to NRI with
+relative ease, by implementing the necessary 'Domain's for those (within
+containerd).
+
+## Enabling NRI Support in Containerd
+
+Enabling and disabling NRI support in containerd happens by enabling or
+disabling the common containerd NRI plugin. The plugin, and consequently
+NRI functionality, is disabled by default. It can be enabled by adding
+a configuration fragment like this to the containerd configuration:
+
+```toml
+  [plugins."io.containerd.nri.v1.nri"]
+    disable = false
+```
+
+In addition to this, you need to put a runtime agnostic NRI configuration
+file in place, to let NRI itself know that it is enabled. You can do this
+by creating `/etc/nri/nri.conf` with the following content:
+
+```yaml
+disableConnections: false
+```
+
+This will enable externally launched NRI plugins to connect and register
+themselves with NRI.
```

Comment:
this needs a bit more explanation... to describe how the plugin connects over grpc/ttrpc 

maybe will be shown in the pic?

Reply:
I added a few paragraphs explaining more what that configuration setting controls, how, and why.